### PR TITLE
DM-25838:  Add unit tests for all measurement tasks

### PR DESCRIPTION
**Pull request by SimonKrughoff**

### CODE REVIEW

File: tests/data/AB1_design_expected_0_70_r.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+blobs:
+- !<lsst.verify.Blob>
+  data: {}
+  identifier: 9b79b6610f3441188d7395fdc5d50809
+  name: AB1_design
+identifier: d42026f20b544bf49aec380efca3b583
+metric: AB1_design
+notes: {}
+unit: marcsec
+value: 7.965623934179928
```
Comment (bechtol) [inline tests/data/AB1_design_expected_0_70_r.yaml:11]: Is it expected that the reference values for r and i band are the same for AB1?

File: tests/testEllipKPM.py [hunk 2]
```diff
@@ -48,42 +49,26 @@ def setUpClass(cls):
...
-
-    def test_am1_hist(self):
-        """Test calculation of am1 on a known catalog."""
-        config = AMxTask.ConfigClass()
-        config.annulus_r = 5.0  # This is what makes it AM1
-        task = AMxTask(config=config)
+    def test_te1(self):
+        """Test calculation of TE1 on a known catalog."""
+        config = TExTask.ConfigClass()
+        # This is what maked it TE1
```
Comment (bechtol) [inline tests/testEllipKPM.py:64]: Typo in comment: maked -> makes

**Review by bechtol — APPROVED** [review]
I appreciate the organization of this PR and the overall approach taken for the unit tests.

Partly for my own understanding, and also for future reference, the `README` in the `tests/data` directory refers to copying the persisted matched catalogs and `Measurement` objects. I am guessing that these files were located by searching through the repo?

Setting the random seed for the photometry metrics seems a good solution for the unit test. 

The tests for setting the configuration seems a good solution as well. 

One more substantial comment: should we consider including a test for the assembly of the matched catalog? If we were to change the way that the matched catalog assembly is done (e.g., filtering), I don't think the current suite of unit tests would catch the change because they read in a copied persisted catalog as input. I think this arrangement is a good choice because it isolates the algorithmic part of computing the metric. I'm thinking that the check of matched catalog assembly would be a separate unit test.

File: tests/data/AB1_design_expected_0_70_r.yaml (hunk 1 above)
Comment (SimonKrughoff) [reply tests/data/AB1_design_expected_0_70_r.yaml:11]: This is a good question.  The way AB1 is defined, I'd expect the r-band value to be zero.

**Comment by SimonKrughoff** [comment]
> Partly for my own understanding, and also for future reference, the README in the tests/data directory refers to copying the persisted matched catalogs and Measurement objects. I am guessing that these files were located by searching through the repo?

That's right.  I just looked through the repo for patches that had measurements for all the metrics I wanted to test.  This was a pretty manual process, but it only needed to be done once, so didn't seem too onerous to me.  As you may notice from the commit history, it took a couple of tries to land on patch 70 which does have data for everything.

> One more substantial comment: should we consider including a test for the assembly of the matched catalog?

I do think we should test the matching, but since this ticket was specifically for measurements, I'd rather put that in another ticket that doesn't block overall adoption of this as the replacement for validate_drp.

**Comment by bechtol** [comment]
Thanks @SimonKrughoff . I agree with the idea to put the unit test for matching on another ticket. I'd say fine to merge.

File: tests/data/AB1_design_expected_0_70_r.yaml (hunk 1 above)
Comment (bechtol) [reply tests/data/AB1_design_expected_0_70_r.yaml:11]: Thanks. Sounds good to check this on a separate ticket.

**Pull request merged**